dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -139,4 +159,100 @@ public CollectionResource wrapResource(CollectionRest collection, String... rels
         return new CollectionResource(collection, utils, rels);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected CollectionRest createAndReturn(Context context) throws AuthorizeException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        CollectionRest collectionRest;
+        try {
+            ServletInputStream input = req.getInputStream();
+            collectionRest = mapper.readValue(input, CollectionRest.class);
+        } catch (IOException e1) {
+            throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
+        }
+
+        Collection collection;
+
+
+        try {
+            Community parent = null;
+            if (StringUtils.isNotBlank(req.getParameter("parentCommunity"))) {
+
+                UUID owningCommunityUuid = UUIDUtils.fromString(req.getParameter("parentCommunity"));
+                if (owningCommunityUuid == null) {
+                    throw new BadRequestException("The given owningCommunityUuid was invalid: "
+                            + req.getParameter("parentCommunity"));
+                }
+
+                parent = communityService.find(context, owningCommunityUuid);
+                if (parent == null) {
+                    throw new ResourceNotFoundException("Parent community for id: "
```

Comment:
Sidenote (not a required change) for future development (please let your developers know).  Now that we use log4j v2, you can now pass in parameters to error messages.  For example, this line could be changed to:

`throw new ResourceNotFoundException("Parent community for id {} not found", parentCommunityUuid);`